src/otx/core/model/entity/instance_segmentation.py

```diff
@@ -127,22 +134,101 @@ def _export_parameters(self) -> dict[str, Any]:
+
+        if isinstance(results_list, list) and isinstance(results_list[0], InstanceData):  # rewrite
+            # Predict case, consists of InstanceData
+            predictions = self.add_pred_to_datasample(data_samples, results_list)
+
+            features_for_sal_map = [data_sample.pred_instances for data_sample in data_samples]
+            saliency_map = self.explain_fn(features_for_sal_map)
+            feature_vector = torch.empty(1, dtype=torch.uint8)
+
+        elif isinstance(results_list, tuple) and isinstance(results_list[0], torch.Tensor):  # rewrite
+            # Export case, consists of tensors
+            predictions = results_list
+
+            saliency_map = torch.empty(1, dtype=torch.uint8)
+            feature_vector = [torch.nn.functional.adaptive_avg_pool2d(f, (1, 1)) for f in x]
+            feature_vector = torch.cat(feature_vector, 1)
+
+        return {
+            "predictions": predictions,
+            "saliency_map": saliency_map,
+            "feature_vector": feature_vector,
+        }
+
+    def get_explain_fn(self) -> Callable:
+        """Returns explain function."""
         from otx.algo.hooks.recording_forward_hook import MaskRCNNRecordingForwardHook
 
-        self.explain_hook = MaskRCNNRecordingForwardHook.create_and_register_hook(
+        explainer = MaskRCNNRecordingForwardHook.create_and_register_hook(
             num_classes=self.num_classes,
         )
+        return explainer.func
+
+    def _reset_model_forward(self) -> None:
+        if not self.explain_mode:
+            return
+
+        self.model.explain_fn = self.get_explain_fn()
+        forward_with_explain = self._forward_explain_inst_seg
 
-    def remove_explain_hook_handle(self) -> None:
-        """Removes explain hook from the model."""
-        if self.explain_hook.handle is not None:
-            self.explain_hook.handle.remove()
+        self.original_model_forward = self.model.forward
 
-    def reset_explain_hook(self) -> None:
-        """Clear all history of explain records."""
-        self.explain_hook.reset()
+        func_type = types.MethodType
+        # Patch class method
+        model_class = type(self.model)
+        model_class.forward = func_type(forward_with_explain, self.model)
+
+    def _restore_model_forward(self) -> None:
+        if not self.explain_mode:
+            return
+
+        if not self.original_model_forward:
+            msg = "Original model forward was not saved."
+            raise RuntimeError(msg)
+
+        func_type = types.MethodType
+        self.model.forward = func_type(self.original_model_forward, self.model)
+        self.original_model_forward = None
+
+    @property
+    def _export_parameters(self) -> dict[str, Any]:
+        """Defines parameters required to export a particular model implementation."""
+        parameters = super()._export_parameters
+        parameters["output_names"] = ["saliency_map", "feature_vector"] if self.explain_mode else None
```

Comment:
All the outputs should be listed here, and probably this doesn't work for mmdeploy when converting to onnx